### DISCUSSION TITLE
Tomcat : io.apiman.manager.api.gateway.GatewayAuthenticationException

### DISCUSSION THREAD

**Discussion by KwanchaiJc** [apiman/apiman#2266] [Help] [answered]
Hi everyone,
I have tried to Configuration Endpoint to public ip / private ip not localhost it works then i go back to my api and i found this error.
How can i solve this problem?
How can i find "Configuration Endpoint Credentials"
![image](https://user-images.githubusercontent.com/85068934/185278864-9f309464-c7d4-4a77-b736-93aa76610c75.png)

io.apiman.manager.api.gateway.GatewayAuthenticationException [Username, Password]
![image](https://user-images.githubusercontent.com/85068934/185279214-7669fdf6-1791-43c3-8ff2-3494c997825d.png)

```
[WARN ] 2022-08-18 09:00:54.771 [http-nio-8080-exec-10] AuthTokenUtil - Missing a Shared-Secret source for auth-token style authentication.  Defaulting to insecure, hard-coded value.

```

**Comment by volkflo** [comment, marked as answer]
Hi @KwanchaiJc 

are you using Keycloak or the Tomcat build-in authentication system?
In the second case you can find the credentials in a file called `tomcat-users.xml`.

**Reply by msavy** [reply]
Thanks, @volkflo  - hopefully your answer is helpful.

@KwanchaiJc - please post stack traces as text, not just as a screenshot.
